### ISSUE TITLE
Retrieve remoteName and RemoteNodeId

### ISSUE THREAD

**Issue by merlinweb** [Ylianst/MeshCentral#130]
Hi, I'm considering to develop a little tool to help routing operations using meshcmd.exe.
With this little program distributed to It support team, we can achieve better performance establishing connections to remote hosts, in the easiest way.
In my example, using port 1433 to connect MMS to Sql Server (in manual mode works like a charm)
It' a quite basic form, see below:
![image](https://user-images.githubusercontent.com/48550288/54447959-f0865c00-474a-11e9-9a66-31de5aa26f56.png)

to make it operational i need to retrieve the remoteName and RemoteNodeId in some way to populate combobox (or searchbox), making simpler to locate a host and configure connection

Is there a way to obtain this information ?

**Comment by jsastriawan** [comment]
See https://github.com/jsastriawan/mc2router.

Feel free to copy/modify/rewrite. Ask me if you need help, I am happy to assist.

Regards,


Joko

**Comment by merlinweb** [comment]
Thanks for your link, but I need a little help about how to use it... :)
My purpose was to coding an interface to fill data in meshaction.txt automatically and then launch meshcmd.exe to establish connection, but your tool is something completely different.
I guess this tool is on the client, but I don't understand where are the configs to adapt it to my server and how to launch index.html correctly to see connection buttons etc i see inside html file...

sorry for my poor knowledge...

**Comment by jsastriawan** [comment]
First of all this software is a NodeWebkit software. Please familiarize yourself with Node Webkit/nwjs. So you will need NodeJS and NW installed on tye device where you want to run it.

The software performs tasks similar to MeshCmd with tunnel action but it is all implemented in NodeJS and NW in GUI.

Once this software is connected to an instance of Meshcentral 2, it will query the list of mesh and nodes. This data will be used to create tunnel similar to meshaction.

Now, if you select one node then click SSH button, the software will create a random local TCP port to listen for a Putty connection and then start Putty to connect to that random port. Once Putty established connection to that random port, the software will initiate Websocket relay tunnel to the MeshAgent of the selected node to port 22. Once the relay is established, TCP traffic between Putty and sshd at the device will flow.

So in short, this software substitutes MeshCmd+meshaction.txt with simple HTML UI.

**Comment by jsastriawan** [comment]
The config file is basically to store the MeshCentral URL and user credential for UI purpose (currently not secure/very basic). Additionally that config file is to store the path of Putty, FileZilla and MS Remote Desktop software that will be used to access the service at the target device.

**Comment by merlinweb** [comment]
Thanks Joko, for your patience and lot of useful info.
Now it's time to begin study and test in nw

Cheers
Merlinweb

**Comment by jsastriawan** [comment]
Hi @merlinweb,  use nw sdk version for easier code debugging and do not hesitate to drop an email to joko(dot)sastriawan(at)intel(dot)com if you need further help privately. We are happy to assist you.

**Comment by merlinweb** [comment]
just downloaded, many thanks, MeshCentral2 is amazing and now I understand why, the team behind it is awesome

Cheers
Merlinweb

**Comment by merlinweb** [comment]
Hi Joko, I' beginning to understand.... Your tool is very interesting:

![image](https://user-images.githubusercontent.com/48550288/54494142-94047780-48d7-11e9-8c74-3c40b419d231.png)

![image](https://user-images.githubusercontent.com/48550288/54494152-a383c080-48d7-11e9-8974-187b96cb57d2.png)

Now I'm trying to modify configurations to assign custom button to SSMS, main mail goal.
I'm a newbie in this enviroment, but seems great !
Thanks for all Joko

Merlinweb

**Comment by jsastriawan** [comment]
Hi @merlinweb, see the 'undefined' button. Actually you can add custom entry by looking at menu 'File' ->'Configure'. You need to understand how randomly generated port are being substituted on the fly. I believe you may not need to modify code at all to achieve your goal. Just add an entry at config file. I will update my documentation on how to use this tool later tonight. Some user has written some document that I haven't added back.
Regards,


Joko

**Comment by merlinweb** [comment]
Just find it and playing with it now, I can confirm your tool is perfect "as is"

**Comment by merlinweb** [comment]
Solved perfectly with a lilttle update to mc2router
thanks Joko !

Merlinweb
